File: src/main/java/net/atos/client/bag/util/IndicatieEnumAdapter.java
/*
 * SPDX-FileCopyrightText: 2023 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.bag.util;

import jakarta.json.bind.adapter.JsonbAdapter;

import nl.info.client.bag.model.generated.Indicatie;

public class IndicatieEnumAdapter implements JsonbAdapter<Indicatie, String> {

    @Override
    public String adaptToJson(final Indicatie indicatie) {
        return indicatie.toString();
    }

    @Override
    public Indicatie adaptFromJson(final String json) {
        return Indicatie.fromValue(json);
    }
}

File: src/main/kotlin/nl/info/client/bag/util/IndicatieEnumAdapter.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.bag.util

import jakarta.json.bind.adapter.JsonbAdapter
import nl.info.client.bag.model.generated.Indicatie

class IndicatieEnumAdapter : JsonbAdapter<Indicatie, String> {
    override fun adaptToJson(indicatie: Indicatie): String = indicatie.toString()
    override fun adaptFromJson(json: String): Indicatie = Indicatie.fromValue(json)
}
